protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/RcptCmdHandler.java

```diff
@@ -210,9 +211,9 @@ private String getContext(SMTPSession session, MailAddress recipientAddress, Str
         } else if (null != recipient) {
             sb.append(" [to:").append(recipient).append(']');
         }
-        if (null != session.getAttachment(SMTPSession.SENDER, State.Transaction)) {
-            MailAddress mailAddress = (MailAddress) session.getAttachment(SMTPSession.SENDER, State.Transaction);
-            sb.append(" [from:").append(mailAddress.asString()).append(']');
+        Optional<MailAddress> sender = (Optional<MailAddress>) session.getAttachment(SMTPSession.SENDER, State.Transaction);
+        if (null != sender && sender.isPresent()) {
```

Comment:
```suggestion
        sender.ifPresent(s -> sb.append(" [from:").append(s.asString()).append(']'));
```

Reply:
if an `Optional` can be null, the problem have to be fixed elsewhere

Reply:
See https://github.com/linagora/james-project/pull/1831#discussion_r227725871